app/src/androidTest/kotlin/ca/rmen/android/poetassistant/ResultListIdlingHelper.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2018 Carmen Alvarez
+ *
+ * This file is part of Poet Assistant.
+ *
+ * Poet Assistant is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Poet Assistant is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Poet Assistant.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package ca.rmen.android.poetassistant
+
+import android.os.Handler
+import android.os.Looper
+import android.support.test.espresso.idling.CountingIdlingResource
+import android.support.v7.widget.RecyclerView
+import android.util.Log
+import ca.rmen.android.poetassistant.main.dictionaries.ResultListAdapter
+
+object ResultListIdlingHelper {
```

Comment:
Looks like much complexity to get tests to pass. Without these changes, many tests fail, as the `RecyclerView`s aren't ready (their data isn't displayed) immediately on the main thread, and `getInstrumentation().waitForIdleSync();` doesn't wait long enough for the UI to be refreshed. The only solution I found that worked was this combination of `CountingIdlingResource`, in combination with a `sleep` (`postDelayed()` when it's decremented)  :(  I tried just adding two `waitForIdleSync()` inside `TestAppUtils.search()`, but tests still failed.